File: src/main/java/pe/nanamochi/banchus/repositories/BeatmapRepository.java
package pe.nanamochi.banchus.repositories;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import pe.nanamochi.banchus.entities.db.Beatmap;

@Repository
public interface BeatmapRepository extends JpaRepository<Beatmap, Integer> {
  Beatmap findById(int id);

  Beatmap findByMd5(String md5);
}

File: banchus-app/src/main/kotlin/pe/nanamochi/banchus/beatmap/repository/BeatmapRepository.kt
package pe.nanamochi.banchus.beatmap.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import pe.nanamochi.banchus.beatmap.entity.Beatmap

@Repository
interface BeatmapRepository : JpaRepository<Beatmap, Int> {
    fun findBeatmapById(id: Int): Beatmap?

    fun findByMd5(md5: String): Beatmap?
}
